state/block_cache.go

```diff
@@ -123,6 +126,44 @@ func (cache *BlockCache) SetStorage(addr Word256, key Word256, value Word256) {
 
 // BlockCache.storage
 //-------------------------------------
+// BlockCache.names
+
+func (cache *BlockCache) GetNameRegEntry(name string) *types.NameRegEntry {
+	entry, removed, _ := cache.names[name].unpack()
+	if removed {
+		return nil
+	} else if entry != nil {
+		return entry
+	} else {
+		entry = cache.backend.GetNameRegEntry(name)
+		cache.names[name] = nameInfo{entry, false, false}
+		return entry
+	}
+}
+
+func (cache *BlockCache) UpdateNameRegEntry(entry *types.NameRegEntry) {
+	name := entry.Name
+	// SANITY CHECK
+	_, removed, _ := cache.names[name].unpack()
+	if removed {
+		panic("UpdateNameRegEntry on a removed name")
```

Comment:
agreed. these are dev based panics tho. like we'll do lots of stress testing and find out if we see these (hopefully not). we can also add a recover statement at the top of ExecTx